arch/Kconfig

```diff
@@ -518,6 +518,7 @@ endmenu # Interrupt configuration
 config INIT_ARCH_HW_AT_BOOT
 	bool "Initialize internal architecture state at boot"
 	depends on ARCH_SUPPORTS_ARCH_HW_INIT
+	imply CACHE_MANAGEMENT
```

Comment:
This seems wrong.  What does resetting architecture state at boot have to do with cache management?  Obviously the answer is going to be "**my** architecture has cache state", but that's not what this (generic) kconfig is for.  Surely you can set this elsewhere?

Reply:
Sure, it can easily be set elsewhere, for example at the SOC level. However, in my opinion this is only a band-aid for the issue that this is intended to fix. #54731 describes a specific issue with USB, but the root cause is not a problem confined to the RT10xx series SOCs.

The root issue here is that commit 34e02946520ff61015b3bf745bb3f7b9544f0004 changes how `z_arm_init_arch_hw_at_boot` functions- previously, this code would run if your SOC enabled `CONFIG_DCACHE`, and used the ARM `SCB_*` cache functions directly. With that commit, this code will not run unless the user enables `CONFIG_CACHE_MANAGEMENT`. This means that a large number of ARM SOCs have their behavior in `z_arm_init_arch_hw_at_boot` changed by this commit (all Cortex-M7 cores will, for example)

Of course, one solution is to enable `CONFIG_CACHE_MANAGEMENT` individually for each ARM SOC that has the functionality. If this is the solution we think is best, I'm okay with that approach, but it seems that we are then leaving SOC vendors out in the cold until they run into a bug caused by the changed behavior here, and realize that `z_arm_init_arch_hw_at_boot` is no longer disabling caches.

Reply:
Sure- enabling `CONFIG_CACHE_MANAGEMENT` at the SOC level makes sense to me, but is the intention to enable it unconditionally? My concern is that we will pull in extra code when it is only needed for these specific subsystems. I suppose that the `sys_cache_*` functions will be discarded by the linker when they aren't called, so this might be a workable solution.